File: spC++/src/Implement/Paragraph.cpp
#include "../Headers/Paragraph.h"
#include "Align.cpp"

Paragraph::Paragraph(std::string pg) {
    this->text = pg;
    alignStrategy = nullptr;
}

Paragraph::Paragraph(Paragraph* paragraph) {
    this->text = paragraph->text;
}

void Paragraph::print() {
    if(alignStrategy != nullptr) {
        alignStrategy->render(text);
    } else {
        (new AlignLeft())->render(text);
    }
}

IElement* Paragraph::clone() {
    return new Paragraph(this);
}

void Paragraph::setAlignStrategy(AlignStrategy *alignStrategy) {
    this->alignStrategy = alignStrategy;
}

File: Java-Labs/lab/src/main/java/sp/lab/Models/Paragraph/Paragraph.java
package sp.lab.Models.Paragraph;

import sp.lab.Models.Element;

public class Paragraph extends Element{
    private String text;
    private IAlignStrategy alignStrategy;

    public Paragraph(String text) {
        this.text = text;
        this.alignStrategy = null;
    }

    public void setAlignStrategy(IAlignStrategy alignStrategy) {
        this.alignStrategy = alignStrategy;
    }

    public void print() {
        if(this.alignStrategy != null) {
            this.alignStrategy.render("Paragraph: " + this.text);
        } else {
            System.out.println("Paragraph: " + this.text);
        }
    }
}
